### ISSUE TITLE
<a> elements containing <div> elements are not working

### ISSUE THREAD

**Issue by bedhub** [OWASP/java-html-sanitizer#52]
When i sanitize links containing sub elements these elements will be moved outside the a element.
```
<a href=\"https://www.xyz.com\"><div>Button text</div></a> 
will result in:
<a href="https://www.xyz.com" rel="nofollow"></a><div>Button text</div>
```

Can you please help me with this issue? 

Example code:
```
		String html = "<a href=\"https://www.xyz.com\"><div>Button text</div></a>";
		PolicyFactory policy = new HtmlPolicyBuilder()
				.allowElements("a", "div")
				.allowUrlProtocols("https")
				.allowAttributes("href").onElements("a")
				.requireRelNofollowOnLinks()
				.toFactory();
		String safeHTML = policy.sanitize(html);
```

**Comment by tyagiakhilesh** [comment]
Hello Bedhub,
Were you able to resolve this issue ? I am facing SIMILAR ( not exact ) kind of problem with table attribute. Any ideas ?

**Comment by bedhub** [comment]
No, i have no solution for this. I would really appreciate if someone can look into this.

**Comment by mikesamuel** [comment]
It looks like [HTML5:"The `<a>` element"](https://www.w3.org/TR/html51/semantics.html#the-a-element) has reclassified changed to use [content model *transparent*](https://www.w3.org/TR/html51/dom.html#transparent) but [`TagBalancingHtmlStreamEventReceiver`](https://github.com/OWASP/java-html-sanitizer/blob/master/src/main/java/org/owasp/html/TagBalancingHtmlStreamEventReceiver.java#L428) still classifies it as *inline*.

One way to solve this and other problems would be to implement handling for content model *transparent* and update the element tables.

**Comment by nosneros** [comment]
Is there a priority list of issue that this could be added to? Considering the change reference from mikesamuel, we will be seeing this more frequently and will have unusable links.

**Comment by jmanico** [comment]
Andy,
Thank you for this patch. We understand how important this is to you. Mike will take a close look as soon as he can. We'll keep this issue alive until it's fully deployed to Maven. 
Aloha + Thank you Andy!,
Jim

**Comment by jmanico** [comment]
Note, this is being worked on here with further discussion on this issue. https://github.com/OWASP/java-html-sanitizer/pull/62

**Comment by mikesamuel** [comment]
https://github.com/OWASP/java-html-sanitizer/tree/issue52 should fix the immediate issue.

**Comment by mikesamuel** [comment]
Fixed at https://github.com/OWASP/java-html-sanitizer/commit/081d403c0b77707a521daad1b35a1b45bf313c85 .  Will push to central.

**Comment by mikesamuel** [comment]
Available at [20160413.1](http://search.maven.org/#artifactdetails%7Ccom.googlecode.owasp-java-html-sanitizer%7Cowasp-java-html-sanitizer%7C20160413.1%7Cjar).
